bindings/python/pymongocrypt/binary.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2019-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Internal helpers for dealing with mongocrypt_binary_t."""
+
+from pymongocrypt.binding import ffi, lib
+from pymongocrypt.errors import MongoCryptError
+
+
+def _to_bytes(mongocrypt_binary):
+    """Returns this mongocrypt_binary_t as bytes."""
+    data = lib.mongocrypt_binary_data(mongocrypt_binary)
+    if data == ffi.NULL:
+        raise MongoCryptError('mongocrypt_binary_data returned NULL')
+    data_len = lib.mongocrypt_binary_len(mongocrypt_binary)
+    return ffi.unpack(ffi.cast("char*", data), data_len)
+
+
+def _write_bytes(mongocrypt_binary, data):
+    """Writes the given data to a mongocrypt_binary_t."""
+    buf = lib.mongocrypt_binary_data(mongocrypt_binary)
+    if buf == ffi.NULL:
+        raise MongoCryptError('mongocrypt_binary_data returned NULL')
+
+    ffi.memmove(buf, data, len(data))
+
+
+class _MongoCryptBinary(object):
+    __slots__ = ("bin",)
+
+    def __init__(self, binary):
+        """Wraps a mongocrypt_binary_t."""
+        self.bin = binary
+        if binary == ffi.NULL:
+            raise MongoCryptError(
+                "unable to create new mongocrypt_binary object")
+
+    def _close(self):
+        """Cleanup resources."""
+        if self.bin is None:
+            return
```

Comment:
I kinda hate returns like this. I think putting everything under "if self.bin is not None" reads better.